ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -768,4 +770,36 @@ private void executeDdlCommand(final DdlCommand ddlCommand) {
       throw new KsqlException(ddlCommandResult.getMessage());
     }
   }
+
+  private void checkPersistentQueryCapacity(
+      final List<PreparedStatement> parsedStatements, final String queriesString) {
+
+    final long numPersistentQueries = parsedStatements.stream().filter(parsedStatement -> {
+      final Statement statement = parsedStatement.getStatement();
+      // Note: RunScript commands also have the potential to create persistent queries,
+      // but we don't count those queries here (to avoid parsing those commands)
+      return statement instanceof CreateAsSelect || statement instanceof InsertInto;
+    }).count();
+
+    if (QueryCapacityUtil.exceedsPersistentQueryCapacity(
+        ksqlEngine,
+        numPersistentQueries,
+        ksqlConfig.getInt(KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG))) {
+      throw new KsqlException(
```

Comment:
Move the exception creation / throwing into `QueryCapacityUtil`. Just add a `throwTooManyActivePersistentQueriesException` method.